x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,214 @@
+package keeper
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	//"cosmossdk.io/errors"
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+
+	autopilottypes "github.com/Stride-Labs/stride/v14/x/autopilot/types"
+	icacallbackstypes "github.com/Stride-Labs/stride/v14/x/icacallbacks/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+const (
+	LiquidStake = "liquidstake"
+	RedeemStake = "redeemstake"
+	NoAction    = "noaction"
+)
+
+// Transfers tokens from the community pool deposit ICA account to the host zone holding module address for that pool
+func (k Keeper) TransferCommunityPoolTokens(ctx sdk.Context, hostZone types.HostZone, token sdk.Coin, autoPilotAction string) error {
+	// Verify that the holding address exists
+	if hostZone.CommunityPoolHoldingAddress == "" || hostZone.CommunityPoolDepositIcaAddress == "" {
+		return errors.New("Invalid holding address or deposit address, cannot build valid ICA commands")
+	}
+
+	// The memo may contain autopilot commands to atomically liquid stake/redeem tokens when transfer succeeds
+	//  both transfer+liquid stake will succeed and tokens will end in the stride side holding address, 
+	//  or neither will and the original base tokens will return to the foreign deposit ICA address
+	memoCommands := ""
+	autopilotMetadata := autopilottypes.RawPacketMetadata{
```

Comment:
I think we may need to specify a redemption address here if I'm not mistaken

Reply:
You mean specifically for the RedeemStake case or even for the LiquidStake?  There is another field which I wasn't initializing in the autopilot struct called "ClaimPacketMetadata" which contains a stride address so perhaps this is where the redemption address is supposed to be included?

Reply:
yeah that's what I meant - but no longer relevant anyways with the new plan!